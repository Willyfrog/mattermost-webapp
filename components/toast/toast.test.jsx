// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {shallowWithIntl, mountWithIntl} from 'tests/helpers/intl-test-helper.jsx';

import Toast from './toast.jsx';

describe('components/Toast', () => {
    const defaultProps = {
        onClick: jest.fn(),
        show: true,
        showOnlyOnce: false,
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(<Toast {... defaultProps}><span>{'child'}</span></Toast>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should click', () => {
        const wrapper = mountWithIntl(<Toast {... defaultProps}><span>{'child'}</span></Toast>);
        const toast = wrapper.find(Toast).instance();

        toast.handleClick();
        expect(defaultProps.onClick).toHaveBeenCalledTimes(1);
    });

    test('should dismiss', () => {
        defaultProps.onDismiss = jest.fn();

        const wrapper = mountWithIntl(<Toast {... defaultProps}><span>{'child'}</span></Toast>);
        const toast = wrapper.find(Toast).instance();

        toast.handleDismiss();
        expect(defaultProps.onDismiss).toHaveBeenCalledTimes(1);
    });

    test('should hide', () => {
        defaultProps.showOnlyOnce = true;

        const wrapper = mountWithIntl(<Toast {... defaultProps}><span>{'child'}</span></Toast>);
        const toast = wrapper.find(Toast).instance();
        expect(toast.state.hide).toEqual(false);
        toast.handleDismiss();
        expect(toast.state.hide).toEqual(true);
    });

    test('should never show', () => {
        defaultProps.showOnlyOnce = true;
        defaultProps.show = false;

        const wrapper = mountWithIntl(<Toast {... defaultProps}><span>{'child'}</span></Toast>);
        const toast = wrapper.find(Toast).instance();
        expect(toast.state.hide).toEqual(true);
    });
});
