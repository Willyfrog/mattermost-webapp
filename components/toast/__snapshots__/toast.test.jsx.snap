// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Toast should match snapshot 1`] = `
<div
  className="toast toast__visible"
  style={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="toast__message"
    onClick={null}
  >
    <OverlayTrigger
      defaultOverlayShown={false}
      delayShow={400}
      overlay={
        <Tooltip
          bsClass="tooltip"
          id="dismissToast1"
          placement="right"
        >
          <div
            className="tooltip__shortcut--txt"
          >
            <FormattedMessage
              defaultMessage="CNTRL+J"
              id="quick_switch_modal.jumpToShortcut.windows"
              values={Object {}}
            />
          </div>
        </Tooltip>
      }
      placement="bottom"
      trigger={
        Array [
          "hover",
          "focus",
        ]
      }
    >
      <div>
        <span>
          child
        </span>
      </div>
    </OverlayTrigger>
  </div>
  <div
    className="toast__dismiss"
    onClick={[Function]}
  >
    <OverlayTrigger
      defaultOverlayShown={false}
      delayShow={400}
      overlay={
        <Tooltip
          bsClass="tooltip"
          id="toast-close__tooltip"
          placement="right"
        >
          <FormattedMessage
            defaultMessage="Close"
            id="general_button.close"
            values={Object {}}
          />
          <div
            className="tooltip__shortcut--txt"
          >
            <FormattedMessage
              defaultMessage="esc"
              id="general_button.esc"
              values={Object {}}
            />
          </div>
        </Tooltip>
      }
      placement="bottom"
      trigger={
        Array [
          "hover",
          "focus",
        ]
      }
    >
      <CloseIcon
        className="close-btn"
        id="dismissToast"
      />
    </OverlayTrigger>
  </div>
</div>
`;
